openmic/vggish/__init__.py

```diff
@@ -23,3 +23,41 @@
 
 __pproc__ = Postprocessor(PCA_PARAMS)
 postprocess = __pproc__.postprocess
+
+
+def waveform_to_features(data, sample_rate, compress=True):
+    '''Converts an audio waveform to VGGish features, with or without
+    PCA compression.
+
+    Parameters
+    ----------
+    data : np.array of either one dimension (mono) or two dimensions (stereo)
+
+    sample_rate:
+        Sample rate of the audio data
+
+    compress : bool
+        If True, PCA and quantization are applied to the features.
+        If False, the features are taken directly from the model output
+
+    Returns
+    -------
+    time_points : np.ndarray, len=n
+        Time points in seconds of the features
+
+    features : np.ndarray, shape=(n, 128)
+        The output features, with or without PCA compression and quantization.
+    '''
+
+    import tensorflow as tf
```

Comment:
what's the rationale for within-function importing here?

Reply:
tf import is heavy (sometimes); we shouldn't do it until called for.

If it's already been loaded elsewhere, then this is free.